contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -662,181 +619,137 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
+        // no deficit
+        if (poolDeficit.n == 0) {
+            return 0;
+        }
 
-        return _removeLiquidityTargetAmount(poolRate, poolAmount, reserveAmount, packedRates, level);
+        return _mulDivF(PPM_RESOLUTION, poolDeficit.n, poolDeficit.d);
     }
 
     /**
-     * @dev returns the amount the provider will receive for removing liquidity
+     * @dev returns the pool deficit based on the total protected amount vs. total
+     * positions value, as a fraction.
+     * note that 0/1 is returned if the pool is not in deficit
      */
-    function _removeLiquidityTargetAmount(
-        Fraction memory poolRate,
-        uint256 poolAmount,
-        uint256 reserveAmount,
-        PackedRates memory packedRates,
-        Fraction memory level
-    ) internal pure returns (uint256) {
-        // get the rate between the reserves upon adding liquidity and now
-        Fraction memory addSpotRate = Fraction({ n: packedRates.addSpotRateN, d: packedRates.addSpotRateD });
-        Fraction memory removeSpotRate = Fraction({ n: packedRates.removeSpotRateN, d: packedRates.removeSpotRateD });
-        Fraction memory removeAverageRate = Fraction({
-            n: packedRates.removeAverageRateN,
-            d: packedRates.removeAverageRateD
-        });
+    function _poolDeficit(IDSToken poolToken)
+        private
+        view
+        returns (Fraction memory)
+    {
+        // get the converter balance
+        IConverter converter = IConverter(payable(_ownedBy(poolToken)));
+        IReserveToken reserveToken = _converterOtherReserve(converter, IReserveToken(address(_networkToken)));
+        uint256 reserveBalance = converter.reserveBalance(reserveToken);
 
-        // calculate the protected amount of reserve tokens plus accumulated fee before compensation
-        uint256 total = _protectedAmountPlusFee(poolAmount, poolRate, addSpotRate, removeSpotRate);
+        // calculate the protected liquidity amount
+        uint256 poolTokenSupply = poolToken.totalSupply();
+        uint256 protectedPoolTokenAmount = poolToken.balanceOf(address(_wallet));
+        uint256 protectedLiquidity = _mulDivF(reserveBalance, protectedPoolTokenAmount, poolTokenSupply);
 
-        // calculate the impermanent loss
-        Fraction memory loss = _impLoss(addSpotRate, removeAverageRate);
+        // get the total positions value
+        uint256 totalValue = totalPositionsValue(poolToken);
 
-        // calculate the compensation amount
-        return _compensationAmount(reserveAmount, Math.max(reserveAmount, total), loss, level);
+        // if the protected liquidity is equal or greater than the total value,
+        // the pool is not in deficit
+        if (protectedLiquidity >= totalValue) {
+            return Fraction({ n: 0, d: 1 });
+        }
+
+        // the pool is in deficit
+        return Fraction({
+            n: totalValue.sub(protectedLiquidity),
```

Comment:
Minor, can use `totalValue - protectedLiquidity` at this point.

Reply:
should never use unchecked operators unless overflow/underflow is desired, the benefit is insignificant, a potential bug can be disastrous.